xbmc/filesystem/AudioBookFileDirectory.cpp

```diff
@@ -55,20 +57,108 @@ bool CAudioBookFileDirectory::GetDirectory(const CURL& url,
+      std::string key = StringUtils::ToUpper(tag->key);
+      // track is matroska's discnumber when at level 50 (these tags) as set by mp3tag
+      // part_number is the matroska spec key
+      if (key == "TRACK" || key == "PART_NUMBER")
+        albumtag.SetDiscNumber(std::stoi(tag->value));
+      else if (key == "SUBTITLE" || key == "SETSUBTITLE")
+        albumtag.SetDiscSubtitle(tag->value);
+      else if (key == "TITLE")
+        albumtag.SetAlbum(tag->value);
+      else if (key == "ALBUM")
+        albumtag.SetAlbum(tag->value);
+      else if (key == "ARTIST")
+        albumtag.SetArtist(tag->value);
+      else if (key == "ARTISTSORT" || key == "ARTIST SORT")
+        albumtag.SetArtistSort(
+            StringUtils::Join(StringUtils::Split(tag->value, separators), musicsep));
+      else if (key == "ALBUMARTIST" || key == "ALBUM ARTIST")
+        albumtag.SetAlbumArtist(
+            StringUtils::Join(StringUtils::Split(tag->value, separators), musicsep));
+      else if (key == "ALBUMARTSTS" || key == "ALBUM ARTISTS")
+        albumtag.SetAlbumArtist(StringUtils::Split(tag->value, separators));
+      else if (key == "ALBUMARTISTSORT" || key == "ALBUM ARTIST SORT")
+        albumtag.SetAlbumArtistSort(
+            StringUtils::Join(StringUtils::Split(tag->value, separators), musicsep));
+      else if (key == "MUSICBRAINZ_ARTISTID")
+        albumtag.SetMusicBrainzArtistID(StringUtils::Split(tag->value, separators));
+      else if (key == "MUSICBRAINZ_ALBUMARTISTID")
+        albumtag.SetMusicBrainzAlbumArtistID(StringUtils::Split(tag->value, separators));
+      else if (key == "MUSICBRAINZ_ALBUMARTIST")
+        albumtag.SetAlbumArtist(tag->value);
+      else if (key == "MUSICBRAINZ_ALBUMID")
+        albumtag.SetMusicBrainzAlbumID(tag->value);
+      else if (key == "MUSICBRAINZ_RELEASEGROUPID")
+        albumtag.SetMusicBrainzReleaseGroupID(tag->value);
+      else if (key == "MUSICBRAINZ_ALBUMSTATUS")
+        albumtag.SetAlbumReleaseStatus(tag->value);
+      else if (key == "MUSICBRAINZ_ALBUMTYPE")
+        albumtag.SetMusicBrainzReleaseType(tag->value);
+      else if (key == "PUBLISHER")
+        albumtag.SetRecordLabel(tag->value);
+      // mp3tag info shows year but the value is stored in date_recorded
+      // equates to TDRC in id3v2.4 ISO 8601 yyyy-mm-dd or part thereof
+      else if (key == "YEAR" || key == "DATE_RECORDED")
+        albumtag.SetReleaseDate(tag->value);
+      else if (key == "ORIGYEAR") // ISO 8601 as above. Equates to TDOR in id3v2.4 (set by mp3tag)
+        albumtag.SetOriginalDate(tag->value);
+      else if (key == "MOOD")
+        albumtag.SetMood(tag->value);
+      // genre could be comma delimited or not. Temporarily add the comma just in case.  true trims
+      // any whitespace around the genre(s)
+      else if (key == "GENRE")
+      {
+        separators.push_back(",");
+        albumtag.SetGenre(StringUtils::Split(tag->value, separators), true);
+        separators.pop_back();
```

Comment:
feels like temp code left in? do artists separators make sense here?

Reply:
No.  Genres are generally singular to a track or album, but that isn't necessarily so.  Multiple genres can be applied to both tracks and albums.  In the case of mp3 id3v4.2 tags multiple genres are supported.  This bit of code just adds the `,` separator temporarily because some software uses it as a delimiter when adding multiple genres.

This is definitely needed and not temp code.